modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -160,10 +178,9 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
             return false;
 #endif
         }
-        else
-            return backendId == DNN_BACKEND_OPENCV ||
-                   (backendId == DNN_BACKEND_HALIDE && haveHalide() &&
-                   (type == MAX || (type == AVE && !pad_t && !pad_l && !pad_b && !pad_r)));
+        return (kernel_size.size() != 3) && (backendId == DNN_BACKEND_OPENCV ||
```

Comment:
```cpp
kernel_size.size() == 2
```

to be more certain

Reply:
In `Layer_Test_ROIPooling.Accuracy` "kernel_size.size() = 0